### ISSUE TITLE
uproot.open crashing with hadded file as input, TypeError: ord() expected string of length 1, but int found

### ISSUE THREAD

**Issue by drdavis** [scikit-hep/uproot#30]
I'm using uproot for handling a few simple ntuples and I've come across an error when trying to open a file produced using `hadd`.
Full Traceback:
```
Python 3.6.3 (default, Oct  4 2017, 06:09:38)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import uproot
>>> uproot.open('test.root')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 59, in open
    return ROOTDirectory.read(localsource(path), **options)
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 138, in read
    classes = _defineclasses(streamerinfos)
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 567, in _defineclasses
    rename = dict((streamerinfo.fName, _safename(streamerinfo.fName)) for streamerinfo in streamerinfos)
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 567, in <genexpr>
    rename = dict((streamerinfo.fName, _safename(streamerinfo.fName)) for streamerinfo in streamerinfos)
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 556, in _safename
    out = re.sub(b"[^a-zA-Z0-9]+", lambda bad: b"_" + b"".join(b"%02x" % ord(x) for x in bad.group(0)) + b"_", name).decode("ascii")
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/re.py", line 191, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 556, in <lambda>
    out = re.sub(b"[^a-zA-Z0-9]+", lambda bad: b"_" + b"".join(b"%02x" % ord(x) for x in bad.group(0)) + b"_", name).decode("ascii")
  File "/Users/ddavis/Software/Python/brewed/envs/fresh/lib/python3.6/site-packages/uproot/rootio.py", line 556, in <genexpr>
    out = re.sub(b"[^a-zA-Z0-9]+", lambda bad: b"_" + b"".join(b"%02x" % ord(x) for x in bad.group(0)) + b"_", name).decode("ascii")
TypeError: ord() expected string of length 1, but int found
```

Here are some toy files (tiny, 2 branches with 2 entries each) created and hadded with ROOT v6.10.08
The individual files that work:
http://cern.ch/ddavis/files/files_for_uproot/out1.root
http://cern.ch/ddavis/files/files_for_uproot/out2.root
The hadded file that doesnt:
http://cern.ch/ddavis/files/files_for_uproot/outs.root

In uproot 2.4.1 (with python 3.6.3) the hadded file causes the crash. Quick google is telling me that ord() is redundant in python3. I'm unfamiliar with this part of python but I'll give it a go and report back if I figure anything out.

**Comment by wdconinc** [comment]
Incidentally(?) same error as #28 ... wonder if related.

**Comment by jpivarski** [comment]
I just want you to know that I'm following this and will be looking into it momentarily. It looks like an easy fix. Right now I'm finishing a major release (2.5.0) and will apply the patch on top of that release. The public API (that which is documented in the Reference of uproot.readthedocs.io) isn't changed.

(Not significantly changed. One argument to "iterate".)

**Comment by jpivarski** [comment]
I just pushed a fix that solves this issue (tested with outs.root) and possibly also #28. The difference between the histogram files was that one had different streamers in it than the other two, including a `vector<string>`. When I generated classes for all the types mentioned in the streamers, I have to mangle names like that because "`<`" and "`>`" can't be used in Python class names. The mangling function had accumulated some cruft of conversions between `str` and `bytes` to handle Python 3, and apparently a file with templates didn't make it into the test suite. (Actually, I ought to add outs.root to the test suite.)
